api-extractor/src/definitions/ApiItem.ts

```diff
@@ -399,6 +411,142 @@ abstract class ApiItem {
+    // belongs to. If it is your own ApiPackage.name/version, then you know it's a local symbol.
+    const currentPackageName: string = this.extractor.package.name;
+
+    const typeName: string = typeReferenceNode.typeName.getText();
+    if (!typeReferencePackagePath || typeReferencePackageName === currentPackageName) {
+      // The type is defined in this project.  Did the person remember to export it?
+      const exportedLocalName: string = this.extractor.package.tryGetExportedSymbolName(currentSymbol);
+      if (exportedLocalName) {
+        // [CASE 1] Local/Exported
+        // Yes; the type is properly exported.
+        // TODO: In the future, here we can check for issues such as a @public type
+        // referencing an @internal type.
+        return;
+      } else {
+        // [CASE 2] Local/Unexported
+        // No; issue a warning
+        this.reportWarning(`The type "${typeName}" needs to be exported by the package`
+          + ` (e.g. added to index.ts)`);
+          return;
+      }
+    }
+
+    // External 
+    // Attempt to load from docItemLoader
+    const scopedPackageName: IScopedPackageName = ApiDefinitionReference.parseScopedPackageName(
+      typeReferencePackageName
+    );
+    const apiDefinitionRefParts: IApiDefinintionReferenceParts = {
+      scopeName: scopedPackageName.scope,
+      packageName: scopedPackageName.package,
+      exportName: '',
+      memberName: ''
+    };
+
+    // the currentSymbol.name is the name of an export, if it contains a '.' then the substring 
+    // after the period is the member name
+    if (currentSymbol.name.indexOf('.') > -1) {
+      const exportMemberName: string[] = currentSymbol.name.split('.');
+      apiDefinitionRefParts.exportName = exportMemberName.pop();
+      apiDefinitionRefParts.memberName = exportMemberName.pop();
+    } else {
+      apiDefinitionRefParts.exportName = currentSymbol.name;
+    }
+
+    const apiDefinitionRef: ApiDefinitionReference = ApiDefinitionReference.createFromParts(
+      apiDefinitionRefParts
+    );
+
+    // Attempt to resolve the type by checking the node modules
+    const resolvedApiItem: ResolvedApiItem = this.extractor.docItemLoader.resolveJsonReferences(
+      apiDefinitionRef,
+      this.reportError
+    );
+
+    if (resolvedApiItem) {
+      // [CASE 3] External/Resolved
+      // This is a reference to a type from an external package, and it was resolved.
+      return;
+    } else {
+      // [CASE 4] External/Unresolved
+      // The type is apparently from an external package, however our heuristic above
+      // wasn't able to resolve the API item. This happens due to limitations of
+      // the typeChecker.getAliasedSymbol() API; we should try to improve it.
+      // We will miss cases here where a member is renamed and we don't handle those 
+      // cases at this moment. 
+      // We ignore these cases for now and reserve the warning for testing
```

Comment:
This comment seems out of date.  What is your plan for handling the case where we can't find a *.api.json file for an external package?  I thought we were going to write a summary of the unresolved package names at the bottom of the *.api.ts file.